cli/azd/internal/repository/app_init.go

```diff
@@ -117,22 +116,52 @@ func (i *Initializer) InitFromApp(
 	}
 	i.console.StopSpinner(ctx, title, input.StepDone)
 
-	isDotNetAppHost := func(p appdetect.Project) bool { return p.Language == appdetect.DotNetAppHost }
-	if idx := slices.IndexFunc(projects, isDotNetAppHost); idx >= 0 {
-		// TODO(ellismg): We will have to figure out how to relax this over time.
-		if len(projects) != 1 {
-			return errors.New("only a single Aspire project is supported at this time")
+	var prjAppHost []appdetect.Project
+	for _, prj := range projects {
+		if prj.Language == appdetect.DotNetAppHost {
+			prjAppHost = append(prjAppHost, prj)
+		}
+	}
+
+	if len(prjAppHost) > 1 {
```

Comment:
nit. Consider braking `InitFromApp()` into more than one single long method.

It's currently hard to see what happens if this condition `if len(prjAppHost) > 1` is false...  the code is hidden and there's a lot going on within this decision. Hopefully we could see smaller logic units to review/spot-possible-bugs easier